checkers/ruleguard_checker.go

```diff
@@ -0,0 +1,96 @@
+package checkers
+
+import (
+	"bytes"
+	"go/ast"
+	"go/token"
+	"io/ioutil"
+	"log"
+
+	"github.com/go-lintpack/lintpack"
+	"github.com/quasilyte/go-ruleguard/ruleguard"
+)
+
+func init() {
+	var info lintpack.CheckerInfo
+	info.Name = "ruleguard"
+	info.Tags = []string{"style", "experimental"}
+	info.Params = lintpack.CheckerParams{
+		"rules": {
+			Value: "",
+			Usage: "path to a gorules file",
+		},
+	}
+	info.Summary = "Detects custom rules violations"
+	info.Details = "Reads a rules file and turns them into go-critic checkers."
+	info.Before = `N/A`
+	info.After = `N/A`
+	info.Note = "See https://github.com/quasilyte/go-ruleguard."
+
+	collection.AddChecker(&info, func(ctx *lintpack.CheckerContext) lintpack.FileWalker {
+		return newRuleguardChecker(&info, ctx)
+	})
+}
+
+func newRuleguardChecker(info *lintpack.CheckerInfo, ctx *lintpack.CheckerContext) *ruleguardChecker {
+	c := &ruleguardChecker{ctx: ctx}
+	rulesFilename := info.Params.String("rules")
+	if rulesFilename == "" {
+		return c
+	}
+
+	// We can handle initialization errors easier when we make
+	// a transition to the go/analysis framework.
+	//
+	// For now, we log error messages and return a ruleguard checker
+	// with an empty rules set.
+
+	data, err := ioutil.ReadFile(rulesFilename)
+	if err != nil {
+		log.Printf("init ruleguard: %v", err)
+		return c
+	}
+
+	fset := token.NewFileSet()
+	rset, err := ruleguard.ParseRules(rulesFilename, fset, bytes.NewReader(data))
+	if err != nil {
+		log.Printf("init ruleguard: %v", err)
```

Comment:
what about: `ruleguard init error: %#v` ? it's a bit easier to grep then.

Reply:
Fixed.